src/hooks/loadables/useLoadSpendingLimits.ts

```diff
@@ -84,9 +84,9 @@ export const useLoadSpendingLimits = (): AsyncResult<SpendingLimitState[]> => {
 
   const [data, error, loading] = useAsync<SpendingLimitState[] | undefined>(
     () => {
-      if (!provider || !safeLoaded || !safe.modules) return
+      if (!provider || !safeLoaded) return
 
-      return getSpendingLimits(provider, safe.modules, safeAddress, chainId)
+      return safe.modules ? getSpendingLimits(provider, safe.modules, safeAddress, chainId) : Promise.resolve([])
     },
     [provider, safeLoaded, safe.modules?.length, safeAddress, chainId, updateSpendingLimitsTag],
     false,
```

Comment:
Removing the third param for `clearData` also fixes it

Reply:
Because w/o it, it will flash between polls. Spending limits will be briefly `[]`.

Reply:
I don't see flashing between polls anymore. Could have been a problem in the past because we didn't memoize certain objects?

Reply:
I've adjusted `useAsync` to instead clear data on every poll and it works as expected without rendering flashes.